test/unit/ukBankHolidays.test.js

```diff
@@ -0,0 +1,211 @@
+        });
+
+        it('returns false when date is not a scottish or northern-irish bank holiday', async function() {
+            const isBankHoliday = await ukBankHolidays('21-03-2018', countries);
+            expect(isBankHoliday).to.equal(false);
+        });
+
+    });
+
+    describe('england-and-wales, northern-ireland and scotland', () => {
+
+        before(() => {
+            countries =  ['england-and-wales', 'northern-ireland', 'scotland'];
+        });
+
+        it('returns true when date is a scottish bank holiday but not english, welsh, northern-irish', async function() {
+            const isBankHoliday = await ukBankHolidays('30-11-2018', countries);
+            expect(isBankHoliday).to.equal(true);
+        });
+
+        it('returns true when date is a northern-irish bank holiday but not english, welsh or scottish', async function() {
+            const isBankHoliday = await ukBankHolidays('12-07-2018', countries);
+            expect(isBankHoliday).to.equal(true);
+        });
+
+        it('returns true when date is an english, welsh and northern-irish bank holiday but not scottish', async function() {
+            const isBankHoliday = await ukBankHolidays('27-08-2018', countries);
+            expect(isBankHoliday).to.equal(true);
+        });
+
+        it('returns false when date is not an english, welsh, scottish or northern-irish bank holiday', async function() {
+            const isBankHoliday = await ukBankHolidays('21-03-2018', countries);
+            expect(isBankHoliday).to.equal(false);
+        });
+
+    });
+
+    describe('date as a string', () => {
+
+        before(() => {
+            countries =  ['england-and-wales', 'northern-ireland', 'scotland'];
+        });
+
+        it('should return true when date is as string and a bank holiday', async function() {
+            const isBankHoliday = await ukBankHolidays('25-12-2018', countries);
+            expect(isBankHoliday).to.equal(true);
+        });
+
+        it('should return false when date is as string and not a bank holiday', async function() {
+            const isBankHoliday = await ukBankHolidays('21-03-2018', countries);
+            expect(isBankHoliday).to.equal(false);
+        });
+
+    });
+
+    describe('date as a moment', () => {
+
+        before(() => {
+            countries =  ['england-and-wales', 'northern-ireland', 'scotland'];
+        });
+
+        it('should return true when date is as string and a bank holiday', async function() {
+            const isBankHoliday = await ukBankHolidays(moment('25-12-2018', 'DD-MM-YYYY'), countries);
+            expect(isBankHoliday).to.equal(true);
+        });
+
+        it('should return false when date is as string and not a bank holiday', async function() {
+            const isBankHoliday = await ukBankHolidays(moment('21-03-2018', 'DD-MM-YYYY'), countries);
+            expect(isBankHoliday).to.equal(false);
+        });
+
+    });
+
+    describe('errors', () => {
+
```

Comment:
As discussed earlier we require a test for when a user accidentally passes in a date string in the wrong format.